lib/search-results-bar/index.tsx

```diff
@@ -0,0 +1,91 @@
+/**
+ * External dependencies
+ */
+import React, { FunctionComponent, useEffect, useRef } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import IconButton from '../icon-button';
+import ChevronRightIcon from '../icons/chevron-right';
+
+import * as S from '../state';
+
+type StateProps = {
+  selectedSearchMatchIndex: number | null;
+  numberOfMatchesInNote: number;
+};
+
+type DispatchProps = {
+  setSearchSelection: (index: number) => any;
+};
+
+type Props = StateProps & DispatchProps;
+
+const SearchResultsBar: FunctionComponent<Props> = ({
+  selectedSearchMatchIndex: index,
+  numberOfMatchesInNote: total,
+  setSearchSelection,
+}) => {
+  const nextButtonRef = useRef<HTMLButtonElement>();
+  const prevButtonRef = useRef<HTMLButtonElement>();
```

Comment:
what's the difference here between `useRef` and `createRef`? I figured that for adding these to HTML elements we'd use `createRef` (since we're not passing the ref down through anything or using it for a variable.

I'm generally confused on the API for refs…

Reply:
Apparently:

> Notice that with a functional component we are using useRef instead of createRef. If you create a ref using createRef in a functional component, React will create a new instance of the ref on every re-render instead of keeping it between renders.

https://medium.com/trabe/react-useref-hook-b6c9d39e2022

(I changed it over because there was a very informative error message that said something like, "createRef won't work here, did you mean useRef?")

Reply:
ah well actually that makes sense now that I read it